### ISSUE TITLE
PyTorch Scatter 2.0.7 PyTorch Sparse 0.6.9 PyTorch Geometric 2.0.4

### ISSUE THREAD

**Issue by jinxinzheok** [YeHaoran-KL/LoPo-TSP#8]
学长您好，
现在下载scatter等包的网址版本号与您的md中不一致，scatter 2.1.0 sparsee 0.6.15  geometric2.6.1.请问下载这种版本是否仍然可行？

**Comment by henry-yeh** [comment]
这个需要测试一下，最重要的是这些包之间（比如torch scatter和torch版本）是兼容的，它们之间兼容应该没什么问题。

**Comment by jinxinzheok** [comment]
理论上安装对了可以直接执行所有代码是吗？包括evaluation?     我现在一二阶段的训练可以正常启动，但是evaluation阶段报错AttributeError: 'LOCALDataset' object has no attribute 'shape'。

这套代码是否采用了预训练，如果采用了预训练哪个是他的预训练代码？
感谢答复！

**Comment by jinxinzheok** [comment]
evalution阶段，我修改了insertion的代码：
def _to_numpy(arr):    
    if isinstance(arr, torch.Tensor):
        return arr.detach().cpu().numpy()
    if isinstance(arr, list):
        return np.array(arr)
    if hasattr(arr, 'data') and isinstance(arr.data, list):
        # 处理LOCALDataset对象
        tensor_data = torch.stack(arr.data)
        return tensor_data.numpy()
加入处理LOCALDataset的方式，调通了
